shared-module/displayio/__init__.c

```diff
@@ -0,0 +1,80 @@
+#include "shared-bindings/displayio/FourWire.h"
+
+extern displayio_fourwire_obj_t board_display_obj;
+
+void start_region_update(displayio_fourwire_obj_t* display, uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1) {
+    // TODO delegate between different display types
+    displayio_fourwire_start_region_update(display, x0, y0, x1, y1);
+}
+
+void finish_region_update(displayio_fourwire_obj_t* display) {
+    // TODO delegate between different display types
+    displayio_fourwire_finish_region_update(display);
+}
+
+void finish_refresh(displayio_fourwire_obj_t* display) {
+    // TODO delegate between different display types
+    displayio_fourwire_finish_refresh(display);
+}
+
+bool frame_queued(displayio_fourwire_obj_t* display) {
+    // TODO delegate between different display types
+    return displayio_fourwire_frame_queued(display);
+}
+
+bool refresh_queued(displayio_fourwire_obj_t* display) {
+    // TODO delegate between different display types
+    return displayio_fourwire_refresh_queued(display);
+}
+
+bool send_pixels(displayio_fourwire_obj_t* display, uint32_t* pixels, uint32_t length) {
+    // TODO delegate between different display types
+    return displayio_fourwire_send_pixels(display, pixels, length);
+}
+
+void displayio_refresh_display(void) {
+    displayio_fourwire_obj_t* display = &board_display_obj;
+
+    if (!frame_queued(display)) {
+        return;
+    }
+    if (refresh_queued(display)) {
+        // We compute the pixels
+        uint16_t x0 = 0;
+        uint16_t y0 = 0;
+        uint16_t x1 = display->width;
+        uint16_t y1 = display->height;
+        size_t index = 0;
+        //size_t row_size = (x1 - x0);
+        uint16_t buffer_size = 256;
```

Comment:
When I did some tests, I found that a 2K buffer was a good balance where allocating the buffer each frame was not slowing it down too much and sending the data over SPI in chunks was fast enough. How did you arrive at 256?

Reply:
It's intentionally small because its stack allocated and not on the heap. Definitely something to experiment with. Having it row aligned can be helpful if we're scaling the whole thing.

Once we check this PR in we can begin optimizing the internals.